apis/v1alpha2/httproute_types.go

```diff
@@ -111,12 +111,8 @@ type HeaderMatchType = v1beta1.HeaderMatchType
 //     headers are not currently supported by this type.
 //
 // * "/invalid" - "/" is an invalid character
-//
-// +kubebuilder:validation:MinLength=1
-// +kubebuilder:validation:MaxLength=256
-// +kubebuilder:validation:Pattern=`^[A-Za-z0-9!#$%&'*+\-.^_\x60|~]+$`
 // +k8s:deepcopy-gen=false
-type HTTPHeaderName = v1beta1.HTTPHeaderName
+type HTTPHeaderName = v1beta1.HeaderName
```

Comment:
Looks like because of the above change to make the GRPC variant `type GRPCHeaderName v1beta1.HeaderName` instead of its own `string` alias?

Reply:
Yep, this was trying to ensure that our types and validation were consistent since by definition a GRPC Header name is going to have the same validation and constraints as HTTP.